src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -0,0 +1,211 @@
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.http.wire;
+
+import com.jcabi.http.Request;
+import com.jcabi.http.Response;
+import com.jcabi.http.Wire;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Wire that caches requests based on Last-Modified
+ * and If-Modified-Since headers.
+ * @todo If the original request already has an If-Modified-Since header
+ * it should be sent directly.
+ * @author Igor Piddubnyi (igor.piddubnyi@gmail.com)
+ * @version $Id$
+ * @since 1.15
+ */
+public final class LastModifiedCachingWire implements Wire {
+
+    /**
+     * Last-Modified header name.
+     */
+    public static final String LAST_MODIFIED = "Last-Modified";
+
+    /**
+     * If-Modified-Since header name.
+     */
+    private static final String IF_MODIFIED_SINCE = "If-Modified-Since";
+
+    /**
+     * Cache.
+     */
+    private final transient Map<Request, Response> cache;
+
+    /**
+     * Original wire.
+     */
+    private final transient Wire origin;
+
+    /**
+     * Public ctor.
+     * @param wire Original wire
+     */
+    public LastModifiedCachingWire(final Wire wire) {
+        this.origin = wire;
+        this.cache = new ConcurrentHashMap<>();
+    }
+
+    // @checkstyle ParameterNumber (5 lines)
+    @Override
+    public Response send(final Request req, final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rsp;
+        if (method.equals(Request.GET)) {
+            rsp = this.consultCache(
+                req, home, method, headers, content, connect, read
+            );
+        } else {
+            rsp = this.origin.send(
+                req, home, method, headers, content, connect, read
+            );
+        }
+        return rsp;
+    }
+
+    /**
+     * Check cache and update if needed.
+     *
+     * @param req Request
+     * @param home URI to fetch
+     * @param method HTTP method
+     * @param headers Headers
+     * @param content HTTP body
+     * @param connect The connect timeout
+     * @param read The read timeout
+     * @return Response obtained
+     * @throws IOException if fails
+     * @checkstyle ParameterNumber (6 lines)
+     */
+    private Response consultCache(final Request req,
+        final String home,
+        final String method,
+        final Collection<Map.Entry<String, String>> headers,
+        final InputStream content,
+        final int connect,
+        final int read) throws IOException {
+        final Response rspReceived;
```

Comment:
@piddubnyi see http://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html